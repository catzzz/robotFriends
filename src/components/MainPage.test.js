import MainPage from "../components/MainPage";
import { create } from "react-test-renderer";
import React from "react";
import { shallow } from "enzyme";

let wrapper;
let app;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "",
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
  app = create(<MainPage {...mockProps} />);
});

it("renders MainPage without crashing", () => {
  expect(app.toJSON()).toMatchSnapshot();
});

it("fileters Robots", () => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "a",
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
  expect(wrapper.instance().filteredRobots()).toEqual([]);
});

it("fileters Robots correctly", () => {
  const filteredRobots = [
    {
      id: 1,
      name: "Leanne Graham",
      username: "Bret",
      email: "Sincere@april.biz",
    },
  ];
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 1,
        name: "Leanne Graham",
        username: "Bret",
        email: "Sincere@april.biz",
      },
    ],
    searchField: "Leanne",
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
  expect(wrapper.instance().filteredRobots()).toEqual(filteredRobots);
});

it("fileters Robots correctly 2", () => {
  const filteredRobots = [
    {
      id: 1,
      name: "Leanne Graham",
      username: "Bret",
      email: "Sincere@april.biz",
    },
  ];
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 1,
        name: "Leanne Graham",
        username: "Bret",
        email: "Sincere@april.biz",
      },
    ],
    searchField: "Xavier",
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
  expect(wrapper.instance().filteredRobots()).toEqual([]);
});
