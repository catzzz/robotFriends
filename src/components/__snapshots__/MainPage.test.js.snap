// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <div>
    <h1
      className="f1"
    >
      RoboFriends
    </h1>
    <button
      color="red"
      id="counter"
      onClick={[Function]}
    >
      Count: 
      0
    </button>
  </div>
  <div>
    <input
      className="pa3 ba b--green bg-lightest-blue"
      placeholder="search robots"
      type="search"
    />
  </div>
  <div
    style={
      Object {
        "border": "5px solid black",
        "height": "500px",
        "overflowY": "scroll",
      }
    }
  >
    <div />
  </div>
</div>
`;
